p2p: set empty timeouts to small values.

williambanfield opened the pull request:
These timeouts default to 'do not time out' if they are not set. This times up resources, potentially indefinitely. If node on the other side of the the handshake is up but unresponsive, the[ handshake call](https://github.com/tendermint/tendermint/blob/edec79448aa1d62b84683b1b22e12e145dbdda7c/internal/p2p/router.go#L720) will _never_ return.

These are proposed values that have not been validated. I intend to validate them in a production setting.

creachadair commented on node/node.go:
```diff
@@ -748,6 +748,8 @@ func getRouterConfig(conf *config.Config, appClient abciclient.Client) p2p.Route
 		}
 
 	}
+	opts.HandshakeTimeout = 15 * time.Second
```
Are these "small" values? They seem rather long, relative to other p2p timeouts (e.g., the gap between dial attempts, for example).

No objection to the values, I just wasn't clear if we have a rubric or are just setting safety margins. (The latter would also be fine)

williambanfield replied:
Yeah, that's true, these values are not particularly small. Safety margins is a better description for what they are, ensuring that we don't try to handshake a peer indefinitely.

williambanfield commented:
Changed this PR a bit. It appears that these values were intended to be configured in the `config.toml` but the connection was never made. Modified to finish wiring that up instead.

creachadair commented on node/node.go:
```diff
@@ -715,7 +715,9 @@ func loadStateFromDBOrGenesisDocProvider(stateStore sm.Store, genDoc *types.Gene
 func getRouterConfig(conf *config.Config, appClient abciclient.Client) p2p.RouterOptions {
 	opts := p2p.RouterOptions{
-		QueueType: conf.P2P.QueueType,
+		QueueType:        conf.P2P.QueueType,
```
👍🏼
